satellite.py

```diff
@@ -86,10 +85,12 @@
 # finally we wish to choose num_constellations variables. We pick strength of 1
 # because we don't want it to be advantageous to violate the constraint by
 # picking more variables
-bqm.update(dimod.generators.combinations(bqm, num_constellations, strength=1))
+bqm.update(dimod.generators.combinations(bqm.variables, data['num_constellations'], strength=1))
 
-# sample from the bqm using simulated annealing
-sampleset = neal.Neal().sample(bqm, num_reads=100).aggregate()
+if args.solver == 'hss':
+    sampleset = LeapHybridSampler().sample(bqm).aggregate()
+else:
```

Comment:
Better to explicitly check with `elif` here as well, and then under `else` fail with "unrecognized solver".

Reply:
Thanks. I made this change.